indico/htdocs/js/indico/RegistrationForm/registrationform.js

```diff
@@ -205,11 +209,14 @@ ndRegForm.directive('ndRegForm', function($rootScope, url, sortableoptions, regF
                 removeSection: function(section) {
                     regFormFactory.Sections.remove({
                         confId: $rootScope.confId,
-                        sectionId: section.id
-                    }, {}, function(updatedSections) {
-                        regFormFactory.processResponse(updatedSections, {
-                            success: function(updatedSections)  {
-                                $scope.sections = updatedSections;
+                        sectionId: section.id,
+                        confFormId: $rootScope.confFormId
+                    }, {}, function(data) {
+                        regFormFactory.processResponse(data, {
+                            success: function()  {
+                                $scope.sections = $scope.sections.filter(function(obj) {
```

Comment:
The reasoning behind the previous behavior is resilience against concurrent changes. If somebody else is editing the registration form, this is a good moment to refresh the front-end with the most updated information on the server. I don't have any strong opinion on this. Do you have any, @pferreir?